maintenance/wikia/Forum/fixCommentIndexes.php

```diff
@@ -0,0 +1,308 @@
+			->runLoop( $dbr, function ( &$threads, $row ) {
+				$threads[] = $row->parent_comment_id;
+			} );
+
+		// Ensure that we're always sending an array back
+		return empty( $threads ) ? [ ] : $threads;
+	}
+
+	/**
+	 * Get the URL for the wiki being worked on by this script
+	 *
+	 * @return string
+	 */
+	static public function getWikiURL() {
+		$dbName = WikiFactory::IDtoDB( F::app()->wg->CityId );
+		$url = WikiFactory::DBtoDomain( $dbName );
+		return WikiFactory::getLocalEnvURL( 'http://'.$url );
+	}
+}
+
+class BrokenThread {
+
+	// Give a buffer for the new comment index we use in case the thread is actively being commented on
+	const RACE_BUFFER = 100;
+
+	protected $threadId;
+
+	protected $commentCount;
+	protected $currentIndex;
+
+	protected $renumberStart;
+	protected $newWallCount;
+
+	/**
+	 * @param int $threadId
+	 */
+	public function __construct( $threadId ) {
+		$this->threadId = $threadId;
+
+		$this->currentIndex = $this->getCurrentIndex();
+		$this->commentCount = $this->getCommentCount();
+
+		// Start renumbering the duplicates at the current index plus a buffer in case comments are
+		// being made while this script runs
+		$this->renumberStart = $this->currentIndex + self::RACE_BUFFER;
+
+		// Set the start for new comments after this script runs to where we started renumbering plus
+		// the total number of comments we have (worst possible case of all comments being dups)
+		$this->newWallCount = $this->renumberStart + $this->commentCount;
+	}
+
+	/**
+	 * Get the index to be used for the next new comment
+	 *
+	 * @return int
+	 */
+	private function getCurrentIndex() {
+		$index = wfGetWikiaPageProp( WPP_WALL_COUNT, $this->threadId );
+		return empty( $index ) ? 0 : $index;
+	}
+
+	/**
+	 * Get the count of comments made on the thread given by $threadId.  We do this as a separate query rather
+	 * call PHP's count on the comments we get later so that we can quickly update the starting comment
+	 * index before we do anything else.
+	 *
+	 * @return array
+	 */
+	private function getCommentCount() {
+		$dbr = wfGetDB( DB_SLAVE );
+		$count = ( new WikiaSQL )
+			->SELECT( 'count(*)' )->AS_( 'count' )
+			->FROM( 'comments_index' )
+			->WHERE( 'parent_comment_id' )->EQUAL_TO( $this->threadId )
+			->ORDER_BY( 'comment_id' )
+			->runLoop( $dbr, function ( &$count, $row ) {
+				$count = $row->count;
+			});
+
+		// Ensure that we're always sending an array back
+		return empty( $count ) ? 0 : $count;
+	}
+
+	/**
+	 * Fix the comment numbering for the thread given by $this->threadId
+	 */
+	public function fix() {
+		FixCommentIndexes::debug( "* Fixing thread ".$this->threadId );
+
+		$this->setNewWallCount();
+		$this->renumberComments();
+	}
+
+	/**
+	 * Sets the new index to use whenever the next comment is made on the current thread
+	 */
+	private function setNewWallCount() {
+		if ( FixCommentIndexes::isTest() ) {
+			return;
+		}
+
+		wfSetWikiaPageProp( WPP_WALL_COUNT, $this->threadId, $this->newWallCount );
```

Comment:
Are you sure it's not used on thread history page?
http://community.wikia.com/wiki/Thread:925591?action=history
I didn't check that but I expect it can introduce gaps in the history.